### ISSUE TITLE
The forge-provided Slider does not stop when releasing the cursor outside the slider

### ISSUE THREAD

**Issue by PlatinPython** [MinecraftForge/MinecraftForge#8485]
**Minecraft Version:** 1.16.5, 1.18.1

**Forge Version:** 36.2.29, 39.0.88

**Steps to Reproduce:**
1. Create a screen with a slider using the forge-provided `Slider` class (`net.minecraftforge.(fml.)client.gui.widget.Slider`)
2. Open said screen, move the slider and keep holding
3. Move the cursor outside the slider box
4. Release the cursor

[Small mods for ease of reproduction](https://gist.github.com/PlatinPython/70443135e678142f77c5d77e55c2be60)

**Description of issue:**
When one releases the cursor outside of a slider's box the slider does not stop changing it's value, to stop it from moving one has to again click on the slider. This also allows moving multiple sliders at once.
[Video](https://streamable.com/e/9wcp5v)

**Comment by desht** [comment]
The problem here is that the `Slider` widget is tracking the dragged state itself, which is unnecessary (and wrong), although it may well have been valid when the widget was created (Minecraft gui handling has changed a *lot* since then).

The vanilla `AbstractSliderButton` widget is an example of how to do this properly; simply let the owning `ContainerEventHandler` track the dragging state (which it does anyway) - it automatically calls `AbstractWidget#onDrag()` on the slider even if the mouse pointer is moved off the slider, as long as the mouse button is held down.

I can try to do a PR when I get a little time, but the changes required are:
* Delete the `dragging` field and all code which references it from `Slider`
* Delete the now-unnecessary `onRelease()` method
* Add an `onDrag()` method:
```java
    @Override
    protected void onDrag(double mouseX, double mouseY, double dragX, double dragY) {
        this.sliderValue = (mouseX - (this.x + 4)) / (float)(this.width - 8);
        updateSlider();
    }
```
* Probably also get rid of that "String concatenation in loop" warning... string concatenation in loops is bad, m'kay?

**Comment by desht** [comment]
I should also note that removing `dragging` is technically a breaking change, since the field is public, but there's a lot of public stuff in that widget which _really_ shouldn't be.

**Comment by PlatinPython** [comment]
I am currently working on a PR to "fix" this. I created a completely new Slider class and deprecated this one. The new class includes some QoL features which allows for better control of the slider. Will link the PR once done (still gotta do some docs).

**Comment by PlatinPython** [comment]
Will be "fixed" once #8496 is merged.
